Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.cs

```diff
@@ -342,10 +376,18 @@ private void AttemptShoot(EntityUid user, EntityUid gunUid, GunComponent gun)
         var shotEv = new GunShotEvent(user, ev.Ammo);
         RaiseLocalEvent(gunUid, ref shotEv);
 
-        if (userImpulse && TryComp<PhysicsComponent>(user, out var userPhysics))
+        if (userImpulse)
         {
-            if (_gravity.IsWeightless(user, userPhysics))
-                CauseImpulse(fromCoordinates, toCoordinates.Value, user, userPhysics);
+            if (user != null && TryComp<PhysicsComponent>(user, out var userPhysics)) // user impulse
+            {
+                if (_gravity.IsWeightless(user.Value, userPhysics))
+                    CauseImpulse(fromCoordinates, toCoordinates.Value, user.Value, userPhysics);
+            }
+            else if (TryComp<PhysicsComponent>(gunUid, out var gunPhysics)) // gun impulse
+            {
+                if (_gravity.IsWeightless(gunUid, gunPhysics))
+                    CauseImpulse(fromCoordinates, toCoordinates.Value, gunUid, gunPhysics);
+            }
```

Comment:
The change is unnecessary if you treat the user as the gun, and also TryComp already handles null entityuids so the null check isn't required.